pkg/apis/hive/v1/syncset_types.go

```diff
@@ -198,6 +198,13 @@ type SyncSetCommonSpec struct {
 	// Secrets is the list of secrets to sync along with their respective destinations.
 	// +optional
 	Secrets []SecretMapping `json:"secretMappings,omitempty"`
+
+	// OmitAnnotation if true indicates that the usual annotation used by the apply command
+	// will be omitted from the target resources. This results in some loss of functionality
+	// when syncing removal of map entries, but it makes it possible to sync large resources
+	// that otherwise could not be synced.
+	// +optional
+	OmitAnnotation bool `json:"omitAnnotation,omitempty"`
```

Comment:
Can we use an enum instead of a bool? Using a bool locks us in only supporting 2 values. Maybe this field could be something like `ApplyBehavior` with values of "Apply" or "CreateUpdate".

Reply:
If following `oc` commands would you be open to "Create", "Replace", and "Apply"?  Asking to see if we can sneak in some behavior I've wanted and we had some discussion on: one time sync.  If "Create", only do it if the resource doesn't exist.  Yes, it may get deleted and I think then the correct behavior is to re-create it if we do this.  "Replace" would be without any annotation.  "Apply" would be the current (pre PR) behavior.

Reply:
Hmm, definitely worth considering. Let me check with the team. However, in the case of this code, we're not really doing a Replace but an Update (in case it exists) or Create if it doesn't. My understanding is that Replace involves deleting and recreating the resource. Not sure we want to do that periodically for anything.

Reply:
Hmm.. yea we don't need a delete+create, wasn't my intent to ask for that.  Update sound good.